tests/Mocks/ClusterInfoMock.cpp

```diff
@@ -0,0 +1,989 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2023-2023 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Michael Hackstein
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Cluster/ClusterInfo.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/RecursiveLocker.h"
+#include "Basics/ScopeGuard.h"
+#include "Basics/system-functions.h"
+#include "Basics/VelocyPackHelper.h"
+#include "Cluster/AgencyCallback.h"
+#include "Cluster/AgencyCallbackRegistry.h"
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterHelpers.h"
+#include "Cluster/ClusterCollectionCreationInfo.h"
+#include "Cluster/ServerState.h"
+#include "Futures/Utilities.h"
+#include "Logger/LogMacros.h"
+#include "Replication2/ReplicatedLog/AgencySpecificationInspectors.h"
+#include "Utils/Events.h"
+#include "VocBase/LogicalCollection.h"
+#include "Replication2/Methods.h"
+#include "Replication2/StateMachines/Document/DocumentStateMachine.h"
+
+using namespace arangodb;
+using namespace arangodb::cluster;
+
+namespace {
+inline arangodb::AgencyOperation IncreaseVersion() {
```

Comment:
nit: I don't think that any of these functions need to be declared `inline`.
They have static linkage anyway.